esb/plugins/org.wso2.developerstudio.eclipse.gmf.esb.diagram/src/org/wso2/developerstudio/eclipse/gmf/esb/diagram/debugger/launch/tabs/DebuggerConfigTab.java

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright (c) 2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.wso2.developerstudio.eclipse.gmf.esb.diagram.debugger.launch.tabs;
+
+import org.apache.commons.lang.StringUtils;
+import org.eclipse.debug.core.ILaunchConfiguration;
+import org.eclipse.debug.core.ILaunchConfigurationWorkingCopy;
+import org.eclipse.debug.ui.AbstractLaunchConfigurationTab;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.layout.GridData;
+import org.eclipse.swt.layout.GridLayout;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Group;
+import org.eclipse.swt.widgets.Text;
+import org.wso2.developerstudio.eclipse.gmf.esb.diagram.debugger.utils.Messages;
+
+import static org.wso2.developerstudio.eclipse.gmf.esb.diagram.debugger.utils.ESBDebuggerConstants.*;
+
+/**
+ * This class is extended from {@link AbstractLaunchConfigurationTab} and used
+ * to edit/view attributes of ESB Mediation Debugger launch configuration.
+ * 
+ * <p>
+ * This class implements the following life-cycle methods:
+ * 
+ * <p>
+ * <u1>
+ * <li>initializeFrom(ILaunchConfiguration)</li>
+ * <li>performApply(ILaunchConfigurationWorkingCopy)</li>
+ * <li>initializeFrom(ILaunchConfiguration)</li> </u1>
+ * 
+ */
+public class DebuggerConfigTab extends AbstractLaunchConfigurationTab {
+	private Text commandPort;
+	private Text eventPort;
+
+	public DebuggerConfigTab() {
+	}
+
```

Comment:
We don't need to have default public constructor, compiler will provide one for us.